modules/agar/src/main/ts/ephox/agar/assertions/ApproxStructures.ts

```diff
@@ -234,6 +265,22 @@ const assertAttrs = (expectedAttrs: Record<string, StringAssert>, actual: SugarE
   });
 };
 
+const assertAttrsWithSpecifics = (expectedAttrs: Record<string, StringAssert>, actual: SugarElement<Element>) => {
+  const allDefinedAttrs = Obj.keys(expectedAttrs);
+  const actualDefinedAttrs = Obj.keys(Obj.bifilter(Attribute.clone(actual), (_value, key) => !Arr.contains([ 'style', 'class' ], key)).t);
+
+  const isEqual = Arr.equal(
```

Comment:
I thought I saw this comment elsewhere, but it's probably worth making a shared function here for asserting sorted arrays

Reply:
Added a shared function here for asserting sorted arrays.